node/src/frost_sign.rs

```diff
@@ -0,0 +1,576 @@
+use std::collections::btree_map::Iter;
+use crate::primitives::ParticipantId;
+use aes_gcm::aead::rand_core::{CryptoRng, RngCore};
+use anyhow::Context;
+use borsh::{BorshDeserialize, BorshSerialize};
+use cait_sith::protocol::{Action, MessageData, Participant, Protocol, ProtocolError};
+use std::collections::BTreeMap;
+
+/// This implementation is valid in the context where we can choose a Leader (Coordinator)
```

Comment:
@bowenwang1996 you said something about this the last time I mentioned that the cait-sith signature scheme can be optimized to have one party generate a signature rather than all. I forgot what the problem was - do you remember?

Reply:
The leader may not be online. We need a way to handle that